src/System.Management.Automation/engine/parser/SemanticChecks.cs

```diff
@@ -214,15 +235,19 @@ public override AstVisitAction VisitAttribute(AttributeAst attributeAst)
                 string name = namedArg.ArgumentName;
                 if (names.Contains(name))
                 {
-                    _parser.ReportError(namedArg.Extent, () => ParserStrings.DuplicateNamedArgument, name);
+                    _parser.ReportError(namedArg.Extent,
+                        nameof(ParserStrings.DuplicateNamedArgument),
+                        ParserStrings.DuplicateNamedArgument,
+                        name);
                 }
                 else
                 {
                     names.Add(name);
 
                     if (!namedArg.ExpressionOmitted && !IsValidAttributeArgument(namedArg.Argument, constantValueVisitor))
                     {
-                        _parser.ReportError(namedArg.Argument.Extent, GetNonConstantAttributeArgErrorExpr(constantValueVisitor));
+                        (string errorId, string errorMsg) = GetNonConstantAttributeArgErrorExpr(constantValueVisitor);
```

Comment:
No need to assign like this, just do `var error = ..` and then `_parser.ReportError(namedArg.Argument.Extent, error.id, error.msg)`